odoo/addons/base/models/ir_actions_report.py

```diff
@@ -1002,12 +1002,15 @@ def report_action(self, docids, data=None, config=True):
 
         discard_logo_check = self.env.context.get('discard_logo_check')
         if self.env.is_admin() and not self.env.company.external_report_layout_id and config and not discard_logo_check:
-            action = self.env["ir.actions.actions"]._for_xml_id("web.action_base_document_layout_configurator")
-            ctx = action.get('context')
-            py_ctx = json.loads(ctx) if ctx else {}
-            report_action['close_on_report_download'] = True
-            py_ctx['report_action'] = report_action
-            action['context'] = py_ctx
-            return action
+            return self._action_configure_external_report_layout(report_action)
 
         return report_action
+
+    def _action_configure_external_report_layout(self, report_action=None):
+        action = self.env["ir.actions.actions"]._for_xml_id("web.action_base_document_layout_configurator")
+        ctx = action.get('context')
+        py_ctx = json.loads(ctx) if ctx else {}
+        report_action['close_on_report_download'] = True
```

Comment:
Default value can be None as per the function signature, `None['close_on_report_download']` is invalid. Either the function's signature is invalid, either this line is.

Reply:
Indeed, it's a leftover from a previous iteration, nice catch :ok_hand: 
The signature is incorrect.